### DISCUSSION TITLE
An unknown option was passed into libcurl on v8

### DISCUSSION THREAD

**Discussion by vicainelli** [curl/curl#10986] [Q&A] [answered]
Context: I have a step on my pipeline using this image: `alpine/terragrunt:1.3.5` 

Days ago the image  was using curl `7.83.1-r6`

```
"Terragrunt plan & apply": (1/3) Installing curl (7.83.1-r6)
```

But since the v8 was introduced, I started to get this error on this step:

```
Error: failed to execute "bash": curl: (48) An unknown option was passed into libcurl
```

Here's the curl command

```bash
curl -u $ARTIFACTORY_USERNAME:$ARTIFACTORY_PASSWORD "https://artifactory.cmx.com/artifactory/${REPOSITORY}/${NPM_NAMESPACE}/${PROJECT}/-/${NPM_NAMESPACE}/${PROJECT}-${VERSION}.tgz" -o  package.tgz > /dev/null
```

The command works fine on my local, cause I'm using `7.86.0`

**Comment by dfandrich** [comment]
What does the expanded command look like? There are only two options in the
command you show, and they've both been supported forever. Is there a space
and/or dash and/or any other special shell variables in the
$ARTIFACTORY_USERNAME or $ARTIFACTORY_PASSWORD variables?  Since their use is
not double quoted, those would mess up the command. Do you have anything in
your ~/.curlrc file?

**Comment by bagder** [comment]
Also, what does `curl -V` show on that host with the issue?

**Comment by vicainelli** [comment]
@dfandrich The username and password have no space/dashes on these variables.

The command looks like this:

```
curl -u username:pass.wor_d "https://artifactory.cmx.ikea.com/artifactory/repo/@namespace/-/@package-0-1-232.tgz" -o package.tgz
```

And there's nothing on the `~/.curlrc` file.

```
cat: can't open '/builder/home/.curlrc': No such file or directory
```

--- 

@bagder, here's what `curl -V` shows on the pipeline:

```
curl 8.0.1 (x86_64-alpine-linux-musl) libcurl/7.83.1 OpenSSL/1.1.1s zlib/1.2.12 brotli/1.0.9 nghttp2/1.47.0
```

**Reply by bagder** [reply, marked as answer]
See? You're using a new tool with an old lib:

> curl 8.0.1 (x86_64-alpine-linux-musl) **libcurl/7.83.1**

**Comment by vicainelli** [comment]
Yeah, you're right @bagder 

I just had to update the image version. 

Thanks!
